examples/virtual-device-app/android/App/feature/closure/src/main/java/com/matter/virtual/device/app/feature/closure/DoorLockViewModel.kt

```diff
@@ -1,17 +1,104 @@
 import com.matter.virtual.device.app.core.ui.BaseViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
 import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
 import timber.log.Timber
 
 @HiltViewModel
-class DoorLockViewModel @Inject constructor(savedStateHandle: SavedStateHandle) :
-  BaseViewModel(savedStateHandle) {
+class DoorLockViewModel
+@Inject
+constructor(
+  getLockStateFlowUseCase: GetLockStateFlowUseCase,
+  getBatPercentRemainingUseCase: GetBatPercentRemainingUseCase,
+  private val setLockStateUseCase: SetLockStateUseCase,
+  private val sendLockAlarmEventUseCase: SendLockAlarmEventUseCase,
+  private val setBatPercentRemainingUseCase: SetBatPercentRemainingUseCase,
+  private val startMatterAppServiceUseCase: StartMatterAppServiceUseCase,
+  private val isFabricRemovedUseCase: IsFabricRemovedUseCase,
+  savedStateHandle: SavedStateHandle
+) : BaseViewModel(savedStateHandle) {
+
+  private val _lockState: StateFlow<Boolean> = getLockStateFlowUseCase()
+  val lockState: LiveData<Boolean>
+    get() = _lockState.asLiveData()
+
+  private val _batteryStatus: MutableStateFlow<Int> =
+    getBatPercentRemainingUseCase() as MutableStateFlow<Int>
+  val batteryStatus: LiveData<Int>
+    get() = _batteryStatus.asLiveData()
+
+  init {
+    viewModelScope.launch { startMatterAppServiceUseCase(matterSettings) }
+
+    viewModelScope.launch {
+      val isFabricRemoved = isFabricRemovedUseCase().successOr(false)
+      if (isFabricRemoved) {
+        Timber.d("Fabric Removed")
+        onFabricRemoved()
+      }
+    }
+  }
 
   override fun onCleared() {
     Timber.d("onCleared()")
     super.onCleared()
   }
+
+  fun onClickButton() {
+    Timber.d("onClickButton()")
+    viewModelScope.launch {
+      Timber.d("current lockState value = ${_lockState.value}")
+      if (_lockState.value == LOCK_STATE_LOCKED) {
+        Timber.d("set value = unlocked")
+        setLockStateUseCase(LOCK_STATE_UNLOCKED)
+      } else {
+        Timber.d("set value = locked")
+        setLockStateUseCase(LOCK_STATE_LOCKED)
+      }
+    }
+  }
+
+  fun onClickSendLockAlarmEventButton() {
+    Timber.d("onClickSendLockAlarmEventButton()")
+    viewModelScope.launch {
+      if (!_lockState.value) {
+        // if lockState == locked, send alarm event and change the lockState to unlocked
+        sendLockAlarmEventUseCase()
+        setLockStateUseCase(LOCK_STATE_UNLOCKED)
+      }
+    }
+  }
+
+  fun updateBatterySeekbarProgress(progress: Int) {
+    _batteryStatus.value = progress
+  }
+
+  fun updateBatteryStatusToCluster(progress: Int) {
+    Timber.d("progress:$progress")
+    viewModelScope.launch {
+      updateBatterySeekbarProgress(progress)
+      setBatPercentRemainingUseCase(progress)
+    }
+  }
+
+  companion object {
+    const val LOCK_STATE_LOCKED = false
```

Comment:
you are re-defining bools. a `LockState` is not intuitively a bool (not clear if `true` is locked or unlocked just looking at the data definition).

So either make this an enum to make it clear by type or rename the state something like `isLocked` or `isUnlocked` and use that without adding defines for bools.